keps/0020-manual-plan-execution.md

```diff
@@ -0,0 +1,175 @@
+// for implementation details of that look at instance.StartPlanExecution
+planStatus.Status = ExecutionPending
+planStatus.UID = uuid.NewUUID()
+
+instance.Status = newStatus
+client.Status().Update(context.TODO(), instance)
+```
+
+This also counts with the fact that on the server-side we have an admission webhook that won't allow setting a status like that in case there is another plan running. Such update would be rejected.
+
+Although it’s very easy to do this, it’s not very kubernetes idiomatic way of doing things especially because Status should never be updated from a client and it should just capture the state of the object.
+
+For some background, including definition of status subresource by [API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status): `The status summarizes the current state of the object in the system, and is usually persisted with the object by an automated processes but may be generated on the fly. At some cost and perhaps some temporary degradation in behavior, the status could be reconstructed by observation if it were lost.`  
+
+#### Pros
+
+Very easy to implement considering the current state of the code
+
+#### Cons
+
+No way to start a plan with just pure kubectl - this is a reply from sig-cli channel where I posed that question (`It doesn't make sense to update the status manually, it should reflect the current state in the cluster. So no, you cannot set it though kubectl`)
+
+### Expose custom HTTP API from controller-manager to allow receiving commands
+
+Manager binary with controllers (that is run within the pod in the cluster) would in this case have to also expose HTTP API and we’ll have to expose a kubernetes service with this custom API. All clients (CLI/user scripts) will talk to this API every time they want to start a new plan. The API could be as simple as 
+
+HTTP POST `/start?planName=deploy`
+
+The validation whether a plan can be started or not would have to be part of the HTTP endpoint implementation.
+
+#### Pros
+
+The complexity of the implementation is still pretty managable and lower than having a full-featured sub-resource
+
+#### Cons
+
+Have to maintain a separate API (with API documentation)
+It is a non-kubernetes API, so no clients for various languages and no kubectl support (although if we keep the API that simple, almost a CURL can do the trick)
+More complex manager pod deployment with additional service
+How do we manage RBAC?
+
+### Have a separate CRD just for maintaining commands to execute a plan
+
+This would basically mean going half-way back to state pre KEP-18 where we had a separate CRD that was created for every plan that was executed. In this implementation we could have a much leaner CRD capturing only the requests to execute a plan with very lean status that does not duplicate the instance's status. Could be as simple as:
```

Comment:
```suggestion
This would basically mean going half-way back to state pre KEP-18 where we had a separate CR that was created for every plan that was executed. In this implementation we could have a much leaner CRD capturing only the requests to execute a plan with very lean status that does not duplicate the instance's status. Could be as simple as:
```